enhancements/update/cluster-profiles.md

```diff
@@ -105,6 +92,92 @@ the required manifests while keeping a single master source.
 The current installation profile is called `self-managed-high-availability`. All current
 manifests must specify it. Future profiles may choose.
 
+#### Design
+
+A cluster profile is specified to the CVO as an identifier an environment variable.
+
+```
+CLUSTER_PROFILE=[identifier]
+```
+
+For a given cluster, only one cluster profile may be in effect.
+
+The profile is a set-once property and cannot be changed once the cluster has started.
+
+The cluster-version-operator picks the value for the cluster profile according to this order:
+* if it is defined and not empty, the environment variable,
+* otherwise, the default profile `self-managed-high-availability`.
+
+Clusters in a version unaware of the cluster profile must upgrade to the `self-managed-high-availability` profile.
+
+When upgrading, outgoing CVO will forward the cluster profile information to the incoming CVO with the environment variable.
+
+`include.release.openshift.io/[identifier]=true` would make the CVO render this manifest only when `CLUSTER_PROFILE=[identifier]`
```

Comment:
@derekwaynecarr is `include.release.openshift.io` something we claimed as API? I see this used for IBM and high availability something.

Reply:
@mfojtik yes